tasks/test_add_picklist_values.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+from tasks import add_picklist_values
+from unittest import mock
+import responses
+from cumulusci.salesforce_api.tests.metadata_test_strings import deploy_result
+from cumulusci.tasks.salesforce.tests.util import create_task
+#from cumulusci.salesforce_api import soap_envelopes
+
+task_options = {
+    "sobject": "hed__Attribute__c",
+    "field": "hed__Attribute_Type__c",
+    "values": "Test Value" # recordtypes, sorted, otherlast
+}
+
+api_version = "47.0"
+
+class TestAddPicklistValues(unittest.TestCase):
+
+    # to-do tests: record types, sorted, otherlast, bad object, bad field, etc
+    
+    @responses.activate
+    def test_basic(self):
```

Comment:
@jstvz @davisagli I have the basic test here almost working, but I'm hoping one of you might be able to help me out with a failure that I haven't been able to figure out how to resolve:

When I run `pytest`, I get the current error: `AttributeError: 'ApiDeploy' object has no attribute 'process_id'`

```
tasks\test_add_picklist_values.py:85:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\cci\lib\site-packages\cumulusci\core\tasks.py:140: in __call__
    self.result = self._run_task()
tasks\add_picklist_values.py:151: in _run_task
    self._deploy_metadata()
tasks\add_picklist_values.py:306: in _deploy_metadata
    return api()
..\..\cci\lib\site-packages\cumulusci\salesforce_api\metadata.py:60: in __call__
    response = self._get_response()
..\..\cci\lib\site-packages\cumulusci\salesforce_api\metadata.py:153: in _get_response
    envelope = self._build_envelope_status()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <cumulusci.salesforce_api.metadata.ApiDeploy object at 0x059B5650>

    def _build_envelope_status(self):
        if self.soap_envelope_status:
>           return self.soap_envelope_status.format(process_id=self.process_id)
E           AttributeError: 'ApiDeploy' object has no attribute 'process_id'

..\..\cci\lib\site-packages\cumulusci\salesforce_api\metadata.py:99: AttributeError
```

Reply:
@robertbwatson when _process_response_start processes the response to the initial deploy call, it looks for an `id` tag in the response and uses that to set the process id which will be sent in the envelope for subsequent requests to check on the deployment status. I think your mock response for the base_deployment_url request would need to pass in something like `extra="<id>FAKE_ID</id>"`

Reply:
@davisagli This worked, thank you so much! I'd tried something similar and it hadn't worked, so I was close but not quite there. Thanks!